packages/jellyfish-transaction/__tests__/script/dftx/dftx_icxorderbook/ICXSubmitDFCHTLC.test.ts

```diff
@@ -0,0 +1,61 @@
+import { SmartBuffer } from 'smart-buffer'
+import { OP_CODES } from '../../../../src/script'
+import { toBuffer, toOPCodes } from '../../../../src/script/_buffer'
+import { OP_DEFI_TX } from '../../../../src/script/dftx'
+import { CICXSubmitDFCHTLC, ICXSubmitDFCHTLC } from '../../../../src/script/dftx/dftx_icxorderbook'
+import { BigNumber } from '@defichain/jellyfish-json'
+
+it('should bi-directional buffer-object-buffer', () => {
+  const fixtures = [
+    '6a4c514466547833d13ee165fbd163c95d2dff1a36d80d5e342603fb5640e4657719476d313d704900ca9a3b0000000020521a24e5418c971da262215bd30bd79f52611a63e038295b603f64fdc07f95a0050000'
+  ]
+
+  fixtures.forEach(hex => {
+    const stack = toOPCodes(
+      SmartBuffer.fromBuffer(Buffer.from(hex, 'hex'))
+    )
+
+    const buffer = toBuffer(stack)
+    expect(buffer.toString('hex')).toBe(hex)
+    expect((stack[1] as OP_DEFI_TX).tx.type).toBe(0x33)
+  })
+})
+
+const header = '6a4c514466547833' // OP_RETURN(0x1a) OP_PUSHDATA1(0x4c) (length 81 = 0x51) CDfTx.SIGNATURE(0x44665478) CICXSubmitDFCHTLC.OP_CODE(0x33)
```

Comment:
const header = '6a4c514466547833' // OP_RETURN(0x1a) OP_PUSHDATA1(0x4c) (length 81 = 0x51) CDfTx.SIGNATURE(0x44665478) CICXSubmitDFCHTLC.OP_CODE(0x33)
```suggestion
const header = '6a4c514466547833' // OP_RETURN(0x6a) OP_PUSHDATA1(0x4c) (length 81 = 0x51) CDfTx.SIGNATURE(0x44665478) CICXSubmitDFCHTLC.OP_CODE(0x33)
```